client/javascripts/controllers.js

```diff
@@ -4,10 +4,64 @@ app.controller("NutritionController", ['NutritionService', function(NutritionSer
   vm.data;
 
   vm.submit = function(){
-    console.log(form.query);
+    // console.log(vm.form.query);
     NutritionService.getData().then(function(data) {
       vm.data = data.data.parsed_body.report.food;
-      window.data = data.data.parsed_body.report.food;
+      vm.data = NutritionService.getNutrients(vm.data);
+
+      vm.macroData = NutritionService.getMacros(vm.data)
+      window.data = vm.data;
+      vm.macroPieMaker();
     });
   }
+
+  // Display Pie Chart of Macronutrients
+  // Following tutorial from 
+  // http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart
+  // TO DO: Move logic to services.js
+  vm.macroPieMaker = function() {
+    vm.macros = NutritionService.getMacros(data);
+    window.macros = vm.macros;
+
+    var macroColors = ['#ECD078', '#D95B43', '#C02942', '#542437', '#53777A', '#CFF09E', '#A8DBA8', '#3B8686'];
+    
+    var color = d3.scale.ordinal()
+      .range(macroColors); 
+
+    var width = 360;
+    var height = 360;
+    var radius = Math.min(width, height) / 2;
+
+    var svg = d3.select('#macroPie')
```

Comment:
In Angular (and similar frameworks like Ember or React), it's generally a bad idea to work directly on the DOM via queries like this.

What you would typically do, is to extract the code for this DOM manipulation into a directive and make sure you attach to the element that directive manages.